ci/scripts/cpp_build.sh

```diff
@@ -121,7 +121,9 @@ if [ "${ARROW_USE_MESON:-OFF}" = "ON" ]; then
   meson setup \
     --prefix=${MESON_PREFIX:-${ARROW_HOME}} \
     --buildtype=${ARROW_BUILD_TYPE:-debug} \
-    -Dtests=$(meson_boolean ${ARROW_BUILD_TESTS:-OFF}) \
+    -Dauto_features=enabled \
```

Comment:
For CI coverage my thought here is that it makes sense just to turn on any feature option by default in the future, so CI can run against it.  If you disagree this is a good idea then no problem to revert this part

gcs and s3 needed to be explicitly disabled because the current configuration will raise when those are enabled